agentendpoint/config_task.go

```diff
@@ -126,54 +119,57 @@ func (c *configTask) reportContinuingState(ctx context.Context, configState agen
 	if res.GetTaskDirective() == agentendpointpb.TaskDirective_STOP {
 		return errServerCancel
 	}
+	if c.lastProgressState == nil {
+		c.lastProgressState = make(map[agentendpointpb.ApplyConfigTaskProgress_State]time.Time)
+	}
+	c.lastProgressState[configState] = time.Now()
+	return nil
+}
+
+// detectPolicyConflicts checks for managed resource conflicts between a proposed
+// resource and all other resources up to this point, adding to the current set
+// ManagedResources.
+func detectPolicyConflicts(proposed, current *config.ManagedResources) error {
+	// TODO: implement
 	return nil
 }
 
 func (c *configTask) validation(ctx context.Context) {
-	// First populate validation results and internal assignment state.
+	// This is all the managed resources by assignment and policy.
+	globalManagedResources := map[string]map[string]*config.ManagedResources{}
+
+	// Validate each resouce and populate results and internal assignment state.
 	c.assignments = map[string]*assignment{}
 	for i, a := range c.Task.GetConfig().GetConfigAssignments() {
+		ctx = clog.WithLabels(ctx, map[string]string{"config_assignment": a.GetConfigAssignment()})
 		assgnmnt := &assignment{policies: map[string]*policy{}}
 		c.assignments[a.GetConfigAssignment()] = assgnmnt
 		aResult := c.results.GetResults()[i]
-		for i, p := range a.GetPolicies() {
-			plcy := &policy{resources: map[string]*resource{}}
-			assgnmnt.policies[p.GetId()] = plcy
-			pResult := aResult.GetOsPolicyResults().GetResults()[i]
-			for i, r := range p.GetResources() {
-				plcy.resources[r.GetId()] = &resource{}
-				result := pResult.GetResourceResults().GetResults()[i]
-				result.GetExecutionSteps()[validationStepIndex] = &agentendpointpb.ApplyConfigTaskOutput_ResourceResult_ExcecutionStep{
-					Step: &agentendpointpb.ApplyConfigTaskOutput_ResourceResult_ExcecutionStep_Validation{
-						Validation: &agentendpointpb.ApplyConfigTaskOutput_ResourceResult_Validation{
-							Outcome: agentendpointpb.ApplyConfigTaskOutput_ResourceResult_Validation_SKIPPED,
-						},
-					},
-				}
-			}
-		}
-	}
-
-	// Unmarshal all resources.
-	for i, a := range c.Task.GetConfig().GetConfigAssignments() {
-		ctx = clog.WithLabels(ctx, map[string]string{"config_assignment": a.GetConfigAssignment()})
-		assgnmnt := c.assignments[a.GetConfigAssignment()]
-		aResult := c.results.GetResults()[i]
 		for i, p := range a.GetPolicies() {
 			ctx = clog.WithLabels(ctx, map[string]string{"policy_id": p.GetId()})
-			plcy := assgnmnt.policies[p.GetId()]
+			policyMR := &config.ManagedResources{}
```

Comment:
Looks like this field is not getting updated?

Reply:
It's meant to happen in detectPolicyConflicts (logic still to be implemented)